akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala

```diff
@@ -371,9 +372,14 @@ private final class RestartWithBackoffFlow[In, Out](
     override protected def logSource = self.getClass
 
     override protected def startGraph() = {
-      val sourceOut = createSubOutlet(in)
-      val sinkIn = createSubInlet(out)
-      Source.fromGraph(sourceOut.source).via(flowFactory()).runWith(sinkIn.sink)(subFusingMaterializer)
+      val sourceOut: SubSourceOutlet[In] = createSubOutlet(in)
+      val sinkIn: SubSinkInlet[Out] = createSubInlet(out)
+
+      Source.fromGraph(sourceOut.source)
+        .via(StreamUtils.delayCancellation[In](10.millis))
```

Comment:
This really is duct tape. 10 millis sounds very short to me, if there is a gc pause just after cancellation hit for example. I think it can be much more liberal as the tax paid when it isn't a failure is just the stream completing a bit slower, not affecting actual throughput of the stream for example.

Maybe we could pick it up from an attribute in the `delayCancellation` stage instead, so it at least can be tuned from the outside? Also, add a comment saying that this a temporary fix while waiting failures-in-cancellations solution.

Reply:
+1 to an attribute, I didn't want to add a setting or param. I put the details in the commit message but can add a comment also.